docs/raw/push-notification-server.md

```diff
@@ -452,54 +470,52 @@ A `PushNotificationRegister` is used to register with a Push Notification server
 
 A push notification preferences message describes the push notification options and tokens for all the devices associated with `PublicKeyClient`.
 
-`device_tokens`: a list of `PushNotificationPreferences`, one for each device owned by the user.
-`version`: an atomically increasing number identifying the current `PushNotificationPreferences`. Any time anything is changed in the record it MUST be increased by the client, otherwise the request will not be accepted.
+`options`: a list of `PushNotificationOptions`, one for each device owned by the user.
+`version`: an monotonically increasing number identifying the current `PushNotificationPreferences`. Any time anything is changed in the record it MUST be increased by the client, otherwise the request will not be accepted.
 `unregister`: whether the account should be unregistered
+`access_token`: the access token that will be given to clients to send push notifications
 
 #### Data disclosed
 
 - Number of devices with push notifications enabled for a given public key
 - The times a push notification record has been modified by the user
 
-### PushNotificationDeviceToken
+### PushNotificationOptions
 
-`PushNotificationDeviceToken` represent the token and preferences for a given device.
+`PushNotificationOptions` represent the token and preferences for a given device.
 
-`token_type`: the type of token. Currently supported is `APN_TOKEN` for Apple Push 
+`token_type`: the type of token. Currently supported is `APN_TOKEN` for Apple Push
 Notification service and `FIREBASE_TOKEN` for `Firebase`.
 `token`: the actual push notification token sent by `Firebase` or `APN`
 `installation_id`: the [`installation_id`](./2-account.md) of the device
-`preferences`: the push notification preferences for this device.
+`filter_setttings`: the push notification filters for this device.
 
 #### Data disclosed
 
 - Type of device owned by a given user
-- The `FIREBASE` or `APN` push notification token 
+- The `FIREBASE` or `APN` push notification token
 
-### PushNotificationOptions
+### PushNotificationFilterSettings
 
 [//]: (Any of these can be sha3 in order not to store it on the server)
 
 `enabled`: whether the device wants to be sent push notifications
-`allowed_user_list`: a list of `SHA3-256` of public keys hex encoded as a string. If non-empty
-only those public keys will be issued a token.
-`block_user_list`: a list of `SHA3-256` of public keys hex encoded as a string. Any public key 
-in this list MUST not be issued a token.
-`block_chat_list`: a list of `SHA3-256` hashes of chat ids. Any chat id in this list will not trigger a notification.
+`allowed_user_list`: a list of `access_token` encrypted with the key generated
+by the AES key generated by Diffie–Hellman between the publisher and the allowed
```

Comment:
~encrypted with the key generated by the AES key generated~
encrypted with the AES key generated